Assembler.Base/InitialFrameHandler.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+
+namespace Assembler.Base
+{
+    public class InitialFrameHandler<TFrame, TMessage> : BaseFrameHandler<TFrame, TMessage>
+        where TFrame : BaseFrame
+        where TMessage : BaseMessageInAssembly
+    {
+        private readonly ILogger _logger;
+
+        public InitialFrameHandler(ITimeBasedCache<TMessage> cache,
+            IFactory<TFrame, string> identifierFactory, ICreator<TMessage> assembledMessageCreator,
+            IMessageEnricher<TFrame, TMessage> enricher, ILoggerFactory loggerFactory) : base(cache,
+            identifierFactory, enricher, assembledMessageCreator)
+        {
+            _logger = loggerFactory.GetLogger(this);
+        }
+
+        public override void Handle(TFrame frame)
+        {
+            string identifier;
+
+            try
+            {
+                identifier = GetIdentifier(frame);
```

Comment:
I do it because I think it could throw an exception.
I didn't think about the enrichers but I think that whoever implements them should take care of it.
I know how to handle an exception with the Identifier factory, I don't want to use that message, I don't want to assume the same path for other classes.
I can't use it in the Base class because of the logger, it's a bit problematic with abstract types and I found it best to just use the logger in the implementing class.

Reply:
1. "I do it because I think it could throw an exception." - What makes you think like that? This is an abstraction, so you cannot "think". If you want to handle it thats fine, I just dont understand why only this and the other are fine
2. See https://github.com/reverser-gever/Assembler/pull/2#discussion_r399771216 , The logger can (and should) be in the base class

Reply:
This is the check where I can see if the frame I got is valid, that's why the only place I check is here